### PR TITLE
Refactor team stats opponent methods

### PR DESCRIPTION
**Pull request by c-bartell**
Extracted helper methods for favorite_opponent and rival
 - opponent_game_teams
 - opponent_win_stats

Rake and specharness passing

### CODE REVIEW

File: lib/team_statistics.rb [hunk 1]
```diff
@@ -108,52 +108,45 @@ def worst_season(team_id)
...
+  def opponent_game_teams(team_id)
     game_ids = games_to_game_ids(games_by_team_id(team_id))
-    opponent_games = {}
-    teams.each do |team|
-      next if team.team_id == team_id
-
-      opponent_games[team.team_id] = {}
-      opponent_games[team.team_id][:game_data] = game_teams.select do |single_game_stats|
-        ((single_game_stats.team_id == team.team_id) && game_ids.include?(single_game_stats.game_id))
+    opponent_ids = teams.reject do |team|
```
Comment (s-steel) [inline lib/team_statistics.rb:113]: Reject enumerable! Nice, haven't used this yet.

File: lib/team_statistics.rb [hunk 2]
```diff
@@ -108,52 +108,45 @@ def worst_season(team_id)
...
+  def favorite_opponent(team_id)
+    favorite_id = opponent_win_stats(team_id).min_by do |id, data|
+      data[:win_percent]
+    end.first
+
     team_info(favorite_id)['team_name']
   end
 
   def rival(team_id)
-    game_ids = games_to_game_ids(games_by_team_id(team_id))
```
Comment (s-steel) [inline lib/team_statistics.rb:137]: Good refactor.  Looks a lot better.

**Review by s-steel — APPROVED** [review]
Good refactoring and use of different enumerables.

File: lib/team_statistics.rb [hunk 3]
```diff
@@ -108,52 +108,45 @@ def worst_season(team_id)
...
-      opponent_games[team.team_id][:game_data] = game_teams.select do |single_game_stats|
-        ((single_game_stats.team_id == team.team_id) && game_ids.include?(single_game_stats.game_id))
+    opponent_ids = teams.reject do |team|
+      team.team_id == team_id
+    end.map(&:team_id)
+    opponent_ids.reduce({}) do |collector, id|
+      collector[id] = {}
+      collector[id][:game_data] = game_teams.select do |game_team|
+        game_ids.include?(game_team.game_id) && game_team.team_id == id
       end
```
Comment (GregJMitchell) [inline lib/team_statistics.rb:120]: Very nice refactor. Very readable

File: lib/team_statistics.rb [hunk 4]
```diff
@@ -108,52 +108,45 @@ def worst_season(team_id)
...
-    end
-    rival_id = opponent_games.keys.max_by do |team_id|
-      opponent_games[team_id][:win_percent]
-    end
+    rival_id = opponent_win_stats(team_id).max_by do |id, data|
+      data[:win_percent]
+    end.first
 
     team_info(rival_id)['team_name']
   end
```
Comment (GregJMitchell) [inline lib/team_statistics.rb:152]: So concise. Really great job shortening that

**Review by GregJMitchell — APPROVED** [review]
Overall really nice refactor. You cut down on so many lines and made it really really readable

**Pull request merged**